### PR TITLE
#170947581 users are able to chat

### PR DESCRIPTION
**Pull request by Ntare22**
#### What does this PR do?
- implements  users are able to chat
#### Description of Task to be completed?
- add chat.html page
- add chat functionality to the ioMiddleware
#### How should this be manually tested?
- clone this repo
- run `npm install`
- register a user through `/api/v1/auth/register`
- access the chat through `/public/chat.html`
- enter the token of the user in the localStorage by opening the developer tools on your browser and going to Application.
- create a token key and add the token
- start writing messages
#### Any background context you want to provide?
- N/A
#### What are the relevant pivotal tracker stories?
- [#170947581](https://www.pivotaltracker.com/story/show/170947581)
#### Screenshots (if appropriate)
image
#### Questions:

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -50,16 +51,22 @@ export const server = app.listen(port, () => process.stdout.write(`Server is run
...
 
+
 io.use(async (socket, next) => {
   ioMiddleware(socket);
   next();
 });
 
-app.use((req, res, next) => {
+io.use(async (socket, next) => {
+  next();
```
Comment (Baraka-Mugisha) [inline src/index.js]: Could you remove this middleware?

**Review by Baraka-Mugisha — CHANGES_REQUESTED** [review]

File: src/index.js (hunk 1 above)
Comment (Ntare22) [reply src/index.js]: It has been removed!

**Comment by Baraka-Mugisha** [comment]
Could you add a screenshot to the PR showing this feature?

File: src/tests/chat.test.js [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+      },
+      'force new connection': true,
+    });
+  });
+  afterEach(() => {
+    clientSocket.disconnect();
+    clientSocket2.disconnect();
+    clientSocket3.disconnect();
+    sinon.restore();
+    // done();
```
Comment (izzett222) [inline src/tests/chat.test.js]: could you remove commented code

File: public/chat.html [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+            }
+        });
+      socket.on('messageFromServer', (dataFromServer) => {
+        console.log(dataFromServer);
+        socket.emit('messageToServer', {data: 'this is comming from chat.html'})
+      })
+
+      socket.on('connection', (msg) => {
+        console.log(msg);
+      });
```
Comment (izzett222) [inline public/chat.html]: hey @Ntare22 are you using this piece pf code

**Review by izzett222 — CHANGES_REQUESTED** [review]

**Review by izzett222 — DISMISSED** [review]

**Review by Baraka-Mugisha — DISMISSED** [review]

**Review by bbaime98 — DISMISSED** [review]

**Review by Baraka-Mugisha — DISMISSED** [review]

File: src/services/localesServices/locales/en.json [hunk 4]
```diff
@@ -169,5 +169,6 @@
 	"Requests found": "Requests found",
 	"Request found": "Request found",
 	"Requests not found": "Requests not found",
-	"Request not found or not yours to approve": "Request not found or not yours to approve"
-}
+	"Request not found or not yours to approve": "Request not found or not yours to approve",
+	"Request not found or not yours to manage": "Request not found or not yours to manage"
+}
```
Comment (bbaime98) [inline src/services/localesServices/locales/en.json]: add new line please.

**Review by bbaime98 — CHANGES_REQUESTED** [review]

**Review by izzett222 — DISMISSED** [review]

**Comment by bbaime98** [comment]
Use the following to totally remove package-lock: `git rm --cached package-lock.json` then git add, commit and push. Remember to add it first into `gitignore` as well.

**Review by bbaime98 — DISMISSED** [review]
Great work Jim!

**Review by izzett222 — APPROVED** [review]

**Review by Baraka-Mugisha — APPROVED** [review]

**Review by bbaime98 — APPROVED** [review]

**Pull request merged**